### PR TITLE
JsonArray optimizations

### PR DESCRIPTION
**Pull request by magicprinc**
- Better stream() and spliterator() implementations
- ctor with initialCapacity
- better copy ctor

### CODE REVIEW

File: src/main/java/io/vertx/core/json/JsonArray.java [hunk 1]
```diff
@@ -65,16 +66,27 @@ public JsonArray() {
...
+  }
+
   /**
    * Create an instance from a List. The List is not copied.
    *
    * @param list the underlying backing list
    */
-  public JsonArray(List list) {
-    if (list == null) {
+  @SuppressWarnings({"rawtypes", "unchecked"})
```
Comment (vietj) [inline src/main/java/io/vertx/core/json/JsonArray.java:81]: that is an avoidable breaking change, we need to keep the List constructor

**Comment by vietj** [comment]
why this PR is also applying what your other PR about json object contains ?

we are not going to review the same code twice.

**Comment by vietj** [comment]
in other words this PR seems to be focusing on JsonArray, it should not care about JsonObject unless there is a relationship

File: src/main/java/io/vertx/core/json/JsonArray.java (hunk 1 above)
Comment (magicprinc) [reply src/main/java/io/vertx/core/json/JsonArray.java:81]: Does it really break?

I just thought widening is not a problem 🤷‍♀️

**Comment by magicprinc** [comment]
I close this PR, sorry, no time for rebase and force-push - deadline at job.
This PR has no much value. So:
https://github.com/eclipse-vertx/vert.x/pull/5002

**Pull request closed without merging**